public/app/features/dashboard-scene/panel-edit/VizPanelManager.test.tsx

```diff
@@ -56,22 +184,232 @@ describe('VizPanelManager', () => {
+          const vizPanelManager = setupTest();
+          vizPanelManager.activate();
+          await Promise.resolve();
+
+          const panel = vizPanelManager.state.panel;
+
+          expect(panel.state.$timeRange).toBeUndefined();
+
+          vizPanelManager.changeQueryOptions({
+            dataSource: {
+              name: 'grafana-testdata',
+              type: 'grafana-testdata-datasource',
+              default: true,
+            },
+            queries: [],
+            timeRange: {
+              from: '1h',
+            },
+          });
+
+          expect(panel.state.$timeRange).toBeInstanceOf(PanelTimeRange);
+          expect((panel.state.$timeRange?.state as PanelTimeRangeState).timeFrom).toBe('1h');
+
+          vizPanelManager.changeQueryOptions({
+            dataSource: {
+              name: 'grafana-testdata',
+              type: 'grafana-testdata-datasource',
+              default: true,
+            },
+            queries: [],
+            timeRange: {
+              from: '2h',
+            },
+          });
+
+          expect((panel.state.$timeRange?.state as PanelTimeRangeState).timeFrom).toBe('2h');
+        });
+
+        it('should remove PanelTimeRange object on time options cleared', async () => {
+          const vizPanelManager = setupTest();
+          vizPanelManager.activate();
+          await Promise.resolve();
+
+          const panel = vizPanelManager.state.panel;
+
+          expect(panel.state.$timeRange).toBeUndefined();
+
+          vizPanelManager.changeQueryOptions({
+            dataSource: {
+              name: 'grafana-testdata',
+              type: 'grafana-testdata-datasource',
+              default: true,
+            },
+            queries: [],
+            timeRange: {
+              from: '1h',
+            },
+          });
+
+          expect(panel.state.$timeRange).toBeInstanceOf(PanelTimeRange);
+
+          vizPanelManager.changeQueryOptions({
+            dataSource: {
+              name: 'grafana-testdata',
+              type: 'grafana-testdata-datasource',
+              default: true,
+            },
+            queries: [],
+            timeRange: {
+              from: null,
+            },
+          });
+
+          expect(panel.state.$timeRange).toBeUndefined();
+        });
+      });
+
+      describe('max data points and interval', () => {
+        it('max data points', async () => {
+          const vizPanelManager = setupTest();
+          vizPanelManager.activate();
+          await Promise.resolve();
+
+          const dataObj = vizPanelManager.queryRunner;
+
+          expect(dataObj.state.maxDataPoints).toBeUndefined();
+
+          vizPanelManager.changeQueryOptions({
+            dataSource: {
+              name: 'grafana-testdata',
+              type: 'grafana-testdata-datasource',
+              default: true,
+            },
+            queries: [],
+            maxDataPoints: 100,
+          });
```

Comment:
I see in every test, we're forced to send all the options. Would it be worth to allow partial updates of them?

```suggestion
          vizPanelManager.changeQueryOptions({
            maxDataPoints: 100,
          });
```